tools/pubsys/src/aws/validate_ami/mod.rs

```diff
@@ -10,34 +10,33 @@ use crate::aws::client::build_client_config;
 use crate::aws::validate_ami::ami::describe_images;
 use crate::Args;
 use aws_sdk_ec2::{Client as AmiClient, Region};
+use clap::Parser;
 use log::{error, info, trace};
 use pubsys_config::InfraConfig;
 use snafu::ResultExt;
 use std::collections::{HashMap, HashSet};
 use std::fs::File;
 use std::path::PathBuf;
-use structopt::{clap, StructOpt};
 
 /// Validates EC2 images by calling `describe-images` on all images in the file given by
 /// `expected-amis-path` and ensuring that the returned `public`, `ena-support`,
 /// `sriov-net-support`, and `launch-permissions` fields have the expected values.
-#[derive(Debug, StructOpt)]
-#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+#[derive(Debug, Parser)]
 pub(crate) struct ValidateAmiArgs {
     /// File holding the expected amis
-    #[structopt(long, parse(from_os_str))]
+    #[arg(long)]
     expected_amis_path: PathBuf,
 
     /// Optional path where the validation results should be written
-    #[structopt(long, parse(from_os_str))]
+    #[arg(long)]
     write_results_path: Option<PathBuf>,
 
-    #[structopt(long, requires = "write-results-path")]
+    #[arg(long, requires = "write_results_path")]
```

Comment:
Why the change from dashes to underscores?  Does this still work properly?

Reply:
`clap` compares this value to the actual struct property, not the rendered argument name. So with the dashes it would error because there was no property found with that name. It needs to match the actual name on line 32.